src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -372,227 +368,97 @@ private Object classDataValues() {
-            int index = 0;
-            for (ClassData p : classData) {
-                // initialize the static field
-                mv.visitVarInsn(Opcodes.ALOAD, 0);
-                emitIconstInsn(mv, index++);
-                mv.visitMethodInsn(Opcodes.INVOKEINTERFACE, "java/util/List",
-                                   "get", "(I)Ljava/lang/Object;", true);
-                mv.visitTypeInsn(Opcodes.CHECKCAST, p.desc.substring(1, p.desc.length()-1));
-                mv.visitFieldInsn(Opcodes.PUTSTATIC, className, p.name, p.desc);
-            }
-        }
-        mv.visitInsn(Opcodes.RETURN);
-        mv.visitMaxs(2, 1);
-        mv.visitEnd();
-    }
-
-    /*
-     * Low-level emit helpers.
-     */
-    private void emitConst(Object con) {
-        if (con == null) {
-            mv.visitInsn(Opcodes.ACONST_NULL);
-            return;
-        }
-        if (con instanceof Integer) {
-            emitIconstInsn((int) con);
-            return;
-        }
-        if (con instanceof Byte) {
-            emitIconstInsn((byte)con);
-            return;
-        }
-        if (con instanceof Short) {
-            emitIconstInsn((short)con);
-            return;
-        }
-        if (con instanceof Character) {
-            emitIconstInsn((char)con);
-            return;
-        }
-        if (con instanceof Long) {
-            long x = (long) con;
-            short sx = (short)x;
-            if (x == sx) {
-                if (sx >= 0 && sx <= 1) {
-                    mv.visitInsn(Opcodes.LCONST_0 + (int) sx);
-                } else {
-                    emitIconstInsn((int) x);
-                    mv.visitInsn(Opcodes.I2L);
+            clb.withField(p.name, p.desc, new Consumer<FieldBuilder>() {
+                @Override
+                public void accept(FieldBuilder fb) {
+                    fb.withFlags(ACC_STATIC|ACC_FINAL);
                 }
-                return;
-            }
-        }
-        if (con instanceof Float) {
-            float x = (float) con;
-            short sx = (short)x;
-            if (x == sx) {
-                if (sx >= 0 && sx <= 2) {
-                    mv.visitInsn(Opcodes.FCONST_0 + (int) sx);
-                } else {
-                    emitIconstInsn((int) x);
-                    mv.visitInsn(Opcodes.I2F);
-                }
-                return;
-            }
-        }
-        if (con instanceof Double) {
-            double x = (double) con;
-            short sx = (short)x;
-            if (x == sx) {
-                if (sx >= 0 && sx <= 1) {
-                    mv.visitInsn(Opcodes.DCONST_0 + (int) sx);
-                } else {
-                    emitIconstInsn((int) x);
-                    mv.visitInsn(Opcodes.I2D);
-                }
-                return;
-            }
-        }
-        if (con instanceof Boolean) {
-            emitIconstInsn((boolean) con ? 1 : 0);
-            return;
+            });
```

Comment:
Typically a field would only need to build with the access flag.   It would be good for this class to just call:

```
clb.withField(p.name, p.desc, ACC_STATIC|ACC_FINAL);
```

I suggest moving the anonymous inner class to the `ClassBuilder.withField` method taking `int flags` parameter.   A comment in the implementation of `ClassBuilder.withField` to explain that anonymous inner class is used due to bootstrapping issue.

Reply:
yes, fixed, thanks.